ACPs/30-avalanche-warp-x-evm.md

```diff
@@ -0,0 +1,216 @@
+
+Note: this is a conservative estimate since writing to the active state merkle trie is more expensive than storing in a separate key-value database. Additionally, the cost of serving valid signatures is significantly cheaper than serving state sync and bootstrapping requests, so the cost to validators of serving signatures over time is not considered a concern that requires charging a higher gas cost up-front.
+
+##### getBlockchainID
+
+`getBlockchainID` serves an already in-memory value the blockchainID, which is available on the Avalanche Context, so this is charged 2 gas commensurate with other in-memory only operations such as accessing the block number.
+
+##### getVerifiedWarpBlockHash and getVerifiedWarpMessage
+
+`GetVerifiedWarpMessageBaseCost` is charged for serving a warp message (either payload type). This operation is fully in-memory, so it only charges 2 gas.
+
+`GasCostPerWarpMessageBytes` charges 100 gas per byte of the Avalanche Warp Message that is unpacked into a Solidity struct.
+
+## Backwards Compatibility
+
+Existing EVM opcodes and precompiles are not modified by activating Avalanche Warp Messaging in the EVM. This is an additive change to activate a Warp Precompile on the Avalanche C-Chain and can be scheduled for activation in any VM running on Avalanche Subnets that are capable of sending / verifying the specified payload types.
+
+## Reference Implementation
+
+A full reference implementation can be found in Subnet-EVM v0.5.9 [here](https://github.com/ava-labs/subnet-evm/tree/v0.5.9/x/warp).
+
+## Security Considerations
+
+Verifying an Avalanche Warp Message requires reading the source subnet's validator set at the P-Chain height specified in the [Snowman++ Block Extension](https://github.com/ava-labs/avalanchego/blob/v1.10.15/vms/proposervm/README.md#snowman-block-extension). The Avalanche PlatformVM provides the current state of the Avalanche P-Chain and maintains reverse diff-layers in order to compute Subnets' validator sets at historical points in time.
+
+As a result, verifying a historical Avalanche Warp Message that references an old P-Chain height requires applying diff-layers from the current state back to the referenced P-Chain height. As Subnets and the P-Chain continue to produce and accept new blocks, verifying the Warp Messages in historical blocks becomes increasingly expensive.
+
+To efficiently handle historical blocks containing Avalanche Warp Messages, the EVM uses the result bitset encoded in the block to determine the validity of Avalanche Warp Messages without requiring a historical P-Chain state lookup. This is considered secure because the network already verified the Avalanche Warp Messages when they were originally verified and accepted.
+
+## Open Questions
```

Comment:
These sound more like FAQs than Open Questions?

Reply:
Replaced with the open question of how to appropriately charge if at all for validator set lookups.

I originally borrowed from https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/20-ed25519-p2p.md#open-questions as a baseline. @dhrubabasu what do you think of updating this section in ACP-20?